tests/foreign_object/tests.py

```diff
@@ -412,3 +412,28 @@ def test_foreign_object_form(self):
         a2 = form.save()
         self.assertEqual(a.pk, a2.pk)
         self.assertEqual(a2.pub_date, datetime.date(2013, 1, 1))
+
+
+class RestrictedConditionsTests(TestCase):
+
+    def setUp(self):
+        for i in range(10):
+            Number.objects.create()
+
+    def test_restrictions_with_no_joining_columns(self):
+        Bracket.objects.create(low_pass_cutoff=4, high_pass_cutoff=7)
+
+        self.assertListEqual(
+            list(Number.objects.filter(low_bracket=1)
```

Comment:
This needs an order_by clause so that the results are guaranteed to come back in the right order.